File: backend/src/main/java/com/backend/ws/dto/CartDto.java
package com.backend.ws.dto;

import java.util.List;

public class CartDto {

    private Long id;
    private List<BookDto> books;
    private int quantity;
    private String ref;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public List<BookDto> getBooks() {
        return books;
    }

    public void setBooks(List<BookDto> books) {
        this.books = books;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public String getRef() {
        return ref;
    }

    public void setRef(String ref) {
        this.ref = ref;
    }
}

File: frontend/src/app/shared/models/cart-dto.ts
import {BaseDto} from "./base-dto";
import {BookDto} from "./book-dto";

export class CartDto extends BaseDto{
  ref: string;
  quantity: number;
  books: Array<BookDto>

  constructor() {
    super();
    this.ref = "";
    this.books = new Array<BookDto>();
  }
}
